### ISSUE TITLE
Button 'X' to clear an input doesnt' work, doesn't clear else submit the form

### ISSUE THREAD

**Issue by JorgJimz** [olton/Metro-UI-CSS#298]
Hi everybody, I'm using the class "input-control text" and I need a "<button>" tag which has "btn-clear" class, happen that when I Click the button, doesn't clear input else submit the form. I know because I can see that the page loads and later I can see on the direction's bar the GET string. Help.. I'm using this wonderful framework into a JSPproject. Thanks.

**Comment by giuliohome** [comment]
I'm having an issue like this one in a [WebSharper form](https://forums.websharper.com/topic/87118).
From Chrome debugger I suspect that the `change` event on the `input` is not triggered, am I wrong? 
Is the above script something that is still currently to add to the form? Or was it only in an old version of the repo? Or do you have any other suggestions?

**Comment by olton** [comment]
https://sandbox.org.ua/olton/code/gVwGEJf7Zq
In this example, if you press the clear button, the input element fired change event.

**Comment by giuliohome** [comment]
You are right. From the debugger I see that I can add this snippet in my page and I see two function in the event listener, of which one (from websharper) is not triggered by the clear. So I don't understand where is the problem but you seem to prove that it is not in your code. I'm not an expert but but what you are showing is quite clear. I will add this snippet from you to my post in WebSharper forum. Thanks for the moment.

**Comment by giuliohome** [comment]
Ok, do you know why this is not fired when the clear button is used?
```javascript
$("#myInput")[0].addEventListener('input', function (evt) {
    alert("Event fired with value " + this.value);
});
```
Is addEventListener not subscribing the changes done by code? Because this is what the debugger is showing for the WebSharper function.

My [sandbox is here](https://sandbox.org.ua/giuliohome/debug/vblkJfAOwn).

**Comment by olton** [comment]
Clear button trigger `change` event, not `input`

**Comment by giuliohome** [comment]
You are right! I have fixed my sandbox and the question above.
But the point remains IMHO.
If I `addEventListener` a listener for `change`, it is not fired by the clear button, am I wrong?

**Comment by giuliohome** [comment]
Also,
adding 
```javascript
element[0].dispatchEvent(new Event("change"));
```
under these lines (from metro.js 12725)
```javascript
container.on(Metro.events.click, ".input-clear-button", function(){
    var curr = element.val();
    element.val(Utils.isValue(o.defaultValue) ? o.defaultValue : "").trigger('change').trigger('keyup').focus();
```
makes it work

**Comment by olton** [comment]
https://sandbox.org.ua/olton/code/gVwGEJf7Zq
All works fine with `addEventListener`

**Comment by giuliohome** [comment]
I don't see the alert when I click on the "X" of the clear button, even in your sandbox above
Don't you agree?

**Comment by olton** [comment]
Enter text in input field and press clear button

**Comment by giuliohome** [comment]
No enter text field and click out.
You receive the alert because of the text entered.
Then press clear button.
Click out: no alert

**Comment by olton** [comment]
If value not changed, then and an event not trigger

**Comment by giuliohome** [comment]
The value changed.
Again, I try to repeat
I enter a value "blablabla"
I click out
Value changed => alert
Now I click clear button
Value changed from "blablabla" to "" but no alert

**Comment by olton** [comment]
Yes, you right. I fix it in next release

**Comment by giuliohome** [comment]
Awesome, thank you very much! I'm applying the above said `dispatchEvent` workaround in my local script version in the meantime.

**Comment by olton** [comment]
Fixed in 4.2.47
Demo https://sandbox.org.ua/olton/code/gVwGEJf7Zq
You can get it from GitHub from 4.2.47 branch, CDN or wait for a release

**Comment by giuliohome** [comment]
Excellent! Thank you very much!
